core/chains/solana/soltxm/txm.go

```diff
@@ -0,0 +1,114 @@
+package soltxm
+
+import (
+	"context"
+	"fmt"
+
+	solanaGo "github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana"
+	solanaClient "github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	_ services.ServiceCtx = (*Txm)(nil)
+	_ solana.TxManager    = (*Txm)(nil)
+)
+
+// Txm manages transactions for the solana blockchain.
+// simple implementation with no persistently stored txs
+type Txm struct {
+	starter    utils.StartStopOnce
+	lggr       logger.Logger
+	tc         func() (solanaClient.ReaderWriter, error)
+	queue      chan *solanaGo.Transaction
+	stop, done chan struct{}
+	cfg        config.Config
+}
+
+// NewTxm creates a txm. Uses simulation so should only be used to send txes to trusted contracts i.e. OCR.
+func NewTxm(tc func() (solanaClient.ReaderWriter, error), cfg config.Config, lggr logger.Logger) *Txm {
+	lggr = lggr.Named("Txm")
+	return &Txm{
+		starter: utils.StartStopOnce{},
+		tc:      tc,
+		lggr:    lggr,
+		queue:   make(chan *solanaGo.Transaction, 1000), // queue can support 1000 pending txs
+		stop:    make(chan struct{}),
+		done:    make(chan struct{}),
+		cfg:     cfg,
+	}
+}
+
+// Start subscribes to queuing channel and processes them.
+func (txm *Txm) Start(context.Context) error {
+	return txm.starter.StartOnce("solanatxm", func() error {
+		go txm.run()
+		return nil
+	})
+}
+
+func (txm *Txm) run() {
+	defer close(txm.done)
+	ctx, cancel := utils.ContextFromChan(txm.stop)
+	defer cancel()
+	for {
+		select {
+		case tx := <-txm.queue:
+			// TODO: this section could be better optimized for sending TXs quickly
+			// fetch client
+			client, err := txm.tc()
+			if err != nil {
+				txm.lggr.Errorw("failed to get client", "err", err)
+				continue
+			}
+			// process tx
+			sig, err := client.SendTx(ctx, tx)
+			if err != nil {
+				txm.lggr.Criticalw("failed to send transaction", "err", err)
+				continue
+			}
+			txm.lggr.Debugw("successfully sent transaction", "signature", sig)
```

Comment:
I think that would give up the ordering garuntee, is that a problem? i.e. sender queues txs A then B, but they race concurrently and B is actually sent before A.
Otherwise I would only worry about sanity bounding the number of goroutines to keep it from running away.

Reply:
hmmm that could potentially be a problem when submitting to the same feed (for the round/epoch progression), we'd see stale report reverts

but not having it could cause liveness issues across multiple feeds? (a tx to one feed blocks submissions to other feeds)

iirc there's not a concern about nonces since solana doesn't use sequential nonces per transmitter account